dbms/src/Storages/StorageFactory.cpp

```diff
@@ -464,6 +466,23 @@ StoragePtr StorageFactory::get(
 
 		const auto & sharding_key = args.size() == 4 ? args[3] : nullptr;
 
+		/// Check that sharding_key exists in the table and has numeric type.
+		if (sharding_key)
+		{
+			auto ci = std::find_if(columns->begin(), columns->end(), [sharding_key] (const auto & column)
+				{
+					return column.name == sharding_key->getColumnName();
```

Comment:
You could specify arbitary expression for sharding key, like `rand()` or `intHash64(UserID)`.